lib/containers/urls.pm

```diff
@@ -51,64 +51,211 @@ sub get_opensuse_registry_prefix {
     }
 }
 
+our %registries = (
```

Comment:
I would go with
```perl
{
     '<flavor>' => {
               '<version>' => {
                        released => [ARRAY_REF | CODE],
                        staging => [ARRAY_REF | CODE],
                        supported_arch => ARRAY_REF  # <= OPTIONAL, if not given all archs are supported
                     },
               '<version_b>' => {
                      <arch_a> => {
                        released => [ARRAY_REF | CODE],
                        staging => [ARRAY_REF | CODE],
                      }
               }
   '<flavor_b>' => {
...
   }
}
```
CODE will always get a HASH as parameter looking like: `( arch => 'xxx', flavor => 'XXX', version => 'XxX' )` and **have to** return a ARRAY_REF.

I also thought about making the hash keys as regex, but I think this will endup as unreadable as before :)

The distinguished between `flavor->version->arch->released` and `flavor->version->released` can be done by checking for the key `released`...

Reply:
This is, how it could look like if we still would use regex as hash keys
```perl
       '15.3' => {
            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.3" },
            totest_registry   => sub {
                if (check_var('ARCH', 'x86_64') || check_var('ARCH', 'aarch64') || check_var('ARCH', 'ppc64le') || check_var('ARCH', 's390x')) {
                    "registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3";
                } elsif (check_var('ARCH', 'arm')) {
                    "registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3";
                } else {
                    "";
                }
            },
```
will be 
```perl
       '15.3' =>  {
          qr/^x86_64|aarch64|ppc64le$/ => {
              released_registry => [ 'registry.opensuse.org/opensuse/leap:15.3'],
              totest_registry   => ['registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3]'
          'arm' => {
                  released => [ 'registry.opensuse.org/opensuse/leap:15.3'],
                  staging => ['registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3'
        }
}
````

Reply:
**I need to correct my self** 
> CODE will always get a HASH as parameter looking like: ( arch => 'xxx', flavor => 'XXX', version => 'XxX' ) and have to return a ARRAY_REF.
I think we do not need a ARRAY as return value, so replace all the ARRAY_REF with simple SCALAR.

Reply:
regarding regex:
maybe we should not go with full regex, but with multiple strings separated by some delimiter like `|`.

Reply:
*this change is ready for review*
for future reference: 
- We tried to use regexp as key for `<version>`
- @cfconrad provided alternatives https://github.com/cfconrad/os-autoinst-distri-opensuse/commit/4c71ec13b577f4f1246816a6d807faeabc774b97 and https://github.com/cfconrad/os-autoinst-distri-opensuse/blob/play_with_container_urls/containers_urls.pl